arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -1032,26 +990,25 @@ ErrorCode DatabaseFeature::dropDatabase(std::string_view name,
         return true;  // try next DataSource
       };
 
-      vocbase->visitDataSources(
-          visitor);  // acquires a write lock to avoid potential deadlocks
+      // acquires a write lock to avoid potential deadlocks
+      vocbase->visitDataSources(visitor);
 
       if (TRI_ERROR_NO_ERROR != res) {
         return res;
       }
 
-      newLists->_databases.erase(it);
-      newLists->_droppedDatabases.insert(vocbase);
+      next = _databases.make(prev);
+      next->erase(name);
+      _droppedDatabases.insert(vocbase);
     } catch (...) {
-      delete newLists;
       return TRI_ERROR_OUT_OF_MEMORY;
     }
```

Comment:
Now that we do not need to prevent leaks by calling `delete` in the catch block, we can make the whole `try...catch` thing obsolete, and simply let the exception escape from here. That could change the effective error code in case an exception occurs, but it will simplify the code and thus probably should be done now.